packages/instantsearch-ui-components/src/components/Hits.tsx

```diff
@@ -40,31 +93,38 @@ export type HitsClassNames = {
    * Class names to apply to each item element
    */
   item: string | string[];
+  /**
+   * Class names to apply to the banner element
+   */
+  bannerRoot: string | string[];
+  /**
+   * Class names to apply to the banner image element
+   */
+  bannerImage: string | string[];
+  /**
+   * Class names to apply to the banner link element
+   */
+  bannerLink: string | string[];
 };
 
-export function createHitsComponent({ createElement }: Renderer) {
+export function createHitsComponent({ createElement, Fragment }: Renderer) {
   return function Hits<THit extends Hit>(userProps: HitsProps<THit>) {
     const {
       classNames = {},
       hits,
       itemComponent: ItemComponent,
       sendEvent,
       emptyComponent: EmptyComponent,
+      banner,
+      bannerComponent: BannerComponent,
       ...props
     } = userProps;
 
-    if (hits.length === 0 && EmptyComponent) {
-      return (
-        <EmptyComponent
-          className={cx(
-            'ais-Hits',
-            classNames.root,
-            cx('ais-Hits--empty', classNames.emptyRoot),
-            props.className
-          )}
-        />
-      );
-    }
+    const DefaultBannerComponent = createDefaultBannerComponent({
+      createElement,
+      Fragment,
+    });
```

Comment:
If this was moved in the upper scope, between line 110 and 110, this will be called only once instead of at every render.
This way `DefaultBannerComponent` won't be remounted each time as it will have the same definition.

Reply:
Since we're still using `Fragment` for now, `createDefaultBannerComponent()` has been moved up before the return statement.